opensrp-chw-core/src/test/java/org/smartregister/chw/core/fragment/CoreBirthNotificationFragmentTest.java

```diff
@@ -0,0 +1,328 @@
+        verify(coreFpRegisterFragment, Mockito.times(1)).onViewClicked(captor.capture());
+        assertEquals(captor.getValue(), view);
+    }
+
+    @Test
+    public void onSyncCompleteTogglesSyncVisibility() {
+        FetchStatus fetchStatus = Mockito.anyObject();
+        ReflectionHelpers.setField(coreFpRegisterFragment, "syncProgressBar", syncProgressBar);
+        coreFpRegisterFragment.onSyncComplete(fetchStatus);
+        verify(syncProgressBar, Mockito.times(2)).setVisibility(View.GONE);
+    }
+
+    @Test
+    public void testOnViewClickedOpensProfile() {
+        FragmentActivity childRegisterActivity = Robolectric.buildActivity(AppCompatActivity.class).create().resume().get();
+        coreFpRegisterFragment = new MockCoreChildBirthRegisterFragment();
+        Context.bindtypes = new ArrayList<>();
+        Whitebox.setInternalState(coreFpRegisterFragment, "clientsView", clientsView);
+        Whitebox.setInternalState(coreFpRegisterFragment, "presenter", presenter);
+        childRegisterActivity.getSupportFragmentManager().beginTransaction().add(0, coreFpRegisterFragment).commit();
+        when(view.getTag(org.smartregister.family.R.id.VIEW_ID)).thenReturn(CLICK_VIEW_NORMAL);
+        CommonPersonObjectClient client = new CommonPersonObjectClient("12", null, "");
+        client.setColumnmaps(new HashMap<String, String>());
+        when(view.getTag()).thenReturn(client);
+        coreFpRegisterFragment.onViewClicked(view);
+        Intent intent = shadowOf(childRegisterActivity).getNextStartedActivity();
+        assertNotNull(intent);
+    }
+
+    @Test
+    public void testShowNotFoundPopupShowsDialog() {
+        when(coreFpRegisterFragment.getActivity()).thenReturn(coreChildRegisterActivity);
+        when(coreChildRegisterActivity.getFragmentManager()).thenReturn(fragmentManager);
+        when(fragmentManager.beginTransaction()).thenReturn(fragmentTransaction);
+        coreFpRegisterFragment.showNotFoundPopup("1234");
+        verify(fragmentManager).beginTransaction();
+        verify(fragmentTransaction).addToBackStack(null);
+        verify(fragmentTransaction).add(any(NoMatchDialogFragment.class), eq(DIALOG_TAG));
+    }
+
+    @Test
+    public void getDefaultSortQueryCallsPresenterGetSortQuery() {
+        coreFpRegisterFragment.getDefaultSortQuery();
+        verify(coreFpRegisterFragment.presenter(), Mockito.times(1)).getDefaultSortQuery();
+    }
+
+    @Test
+    public void getDueFilterConditionCallsPresenterGetSortQuery() {
+        coreFpRegisterFragment.getDueFilterCondition();
+        verify(coreFpRegisterFragment.presenter(), Mockito.times(1)).getDueFilterCondition();
+    }
+
+    @Test
+    public void testOnViewClickedDoesNothing() {
+        when(coreFpRegisterFragment.getActivity()).thenReturn(null);
+        coreFpRegisterFragment.onViewClicked(view);
+        verifyZeroInteractions(view);
+    }
+
+    @Test
+    public void testRefreshSyncProgressSpinner() {
```

Comment:
Kindly follow the test naming convention `test[MethodName][ExpectedResult][Parameters]`